Functional Programming/Problem2/Problem2.js

```diff
@@ -0,0 +1,29 @@
+(function () {
+  'use strict';
+
+  function curry(fn) {
+    var length = fn.length;
+
+    function wrap(prev) {
+      return function(arg) {
+        var args = prev.concat(arg);
+        if (args.length < length) {
+          return wrap(args);
+        } else {
+          return fn.apply(this, args);
```

Comment:
What do you expect to see in **this**. Try to assign carried function to object and check what is **this**.